src/store/builder/index.ts

```diff
@@ -354,3 +354,22 @@ export function generateConnectionLayoutKeyFor(source: IBlock, target: IBlock) {
     ...map(target.exits, 'tag'),
   ]
 }
+
+export function computeBlockPositionsFrom(block: IBlock) {
```

Comment:
Does this break when null/empty is provided? 
Do we need the type to specific null/undefined?

Reply:
no it won't break.
The only scenario we will get `null` is when there is no activated block. Changed to `IBlock | null`